### PR TITLE
Adapt new `WorkerPoolHash` function in Gardener `v1.98`

### PR DESCRIPTION
**Pull request by Duciwuci**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement
/platform openstack

**What this PR does / why we need it**:

This Pull Requests upgrades the Gardener dependency `v1.97` -> `v1.98`. 
This includes also the new `WorkerPoolHash` function, which now needs an additional variable to be passed.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/2590 in Gardener/Gardener.

**Special notes for your reviewer**:

**Release note**:

```other operator
Adapt new `WorkerPoolHash` function in Gardener `v1.98`.
```

### CODE REVIEW

**Comment by gardener-robot-ci-2** [comment]
Thank you @Duciwuci for your contribution. Before I can start building your PR, a member of the organization must set the required label(s) {'reviewed/ok-to-test'}. Once started, you can check the build status in the PR checks section below.

File: pkg/controller/worker/machines_test.go [hunk 1]
```diff
@@ -358,8 +358,8 @@ var _ = Describe("Machines", func() {
 					},
 				}
 
-				workerPoolHash1, _ = worker.WorkerPoolHash(w.Spec.Pools[0], cluster)
-				workerPoolHash2, _ = worker.WorkerPoolHash(w.Spec.Pools[1], cluster)
+				workerPoolHash1, _ = worker.WorkerPoolHash(w.Spec.Pools[0], cluster, []string{}, []string{})
```
Comment (LucaBernstein) [inline pkg/controller/worker/machines_test.go:361]: any preference of `worker.WorkerPoolHash(w.Spec.Pools[0], cluster, []string{}, []string{})` over `worker.WorkerPoolHash(w.Spec.Pools[0], cluster, nil, nil)`?

Comment (Duciwuci) [reply pkg/controller/worker/machines_test.go]: No, thanks. I adjusted it :)

**Review by AndreasBurger — DISMISSED** [review]
lgtm, thanks!

could you run `make generate` and add the resulting changes to the PR?

**Review by AndreasBurger — APPROVED** [review]

**Pull request merged**